src/librustc_resolve/resolve_imports.rs

```diff
@@ -973,7 +973,7 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
                    max_vis.get() != ty::Visibility::Invisible && // Allow empty globs.
                    !max_vis.get().is_at_least(directive.vis.get(), &*self) {
                     let msg = "A non-empty glob must import something with the glob's visibility";
```

Comment:
This should probably be reworded into something without "must".
Like "this glob import doesn't import anything because all the candidates are too private" or something.
(Also, errors and warnings conventionally start with a lowercase letter.)

Reply:
Maybe the other way "no candidate is public enough" ?

Reply:
>Maybe the other way "no candidate is public enough" ?

Sounds good.

Also, "doesn't import anything" -> "doesn't reexport anything", because the does actually import something, but it is not visible to outer modules as stated.